pkg/kubelet/network/exec/exec.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
+	cmdexec "github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/golang/glog"
+)
+
+type execNetworkPlugin struct {
+	execName string
+	execPath string
+	host     network.Host
+}
+
+const (
+	initCmd     = "--init"
+	setUpCmd    = "--setup"
+	tearDownCmd = "--teardown"
+	execDir     = "/usr/libexec/kubernetes/kubelet-plugins/net/exec/"
+)
+
+func ProbeNetworkPlugins() []network.NetworkPlugin {
+	return ProbeNetworkPluginsWithExecDir(execDir)
+}
+
+func ProbeNetworkPluginsWithExecDir(pluginDir string) []network.NetworkPlugin {
+	execPlugins := []network.NetworkPlugin{}
+
+	files, _ := ioutil.ReadDir(pluginDir)
+	for _, f := range files {
+		// only directories are counted as plugins
+		// and we expect pluginDir/dirname/dirname to be an executable
```

Comment:
Is there a justification for the decision to look for `pluginDir/dirname/dirname` as a binary rather than `pluginDir/binary`?  If so, can we document that maybe as a great big comment at the `package exec` statement above?